addons/common/fnc_removePerFrameHandler.sqf

```diff
@@ -18,34 +18,24 @@ Examples:
     (end)
 
 Author:
-    Nou & Jaynus, donated from ACRE project code for use by the community.
-
+    Nou & Jaynus, donated from ACRE project code for use by the community; commy2
 ---------------------------------------------------------------------------- */
-
 #include "script_component.hpp"
 
-params ["_publicHandle"];
-
-if (isNil "_publicHandle" || {_publicHandle < 0} || {(count GVAR(PFHhandles)) <= _publicHandle}) exitWith {// Nil/no handle or handle is out of bounds of Public Handle Array
-    WARNING("Invalid or not existing PFH ID.");
-};
-
-private "_handle";
-_handle = GVAR(PFHhandles) select _publicHandle;
-if (isNil "_handle") exitWith {}; // Nil handle, nil action
-GVAR(PFHhandles) set [_publicHandle, nil];
-GVAR(perFrameHandlerArray) set [_handle, nil];
-_newArray = [];
-
-GVAR(nextPFHid) = ({
-    private ["_newHandle", "_return"];
-    _return = false;
-    if !(isNil "_x") then {
-        _x params ["", "", "", "", "", "_publicH"];
-        _newHandle = _newArray pushBack _x;
-        GVAR(PFHhandles) set [_publicH, _newHandle];
-        _return = true;
-    };
-    _return
-} count GVAR(perFrameHandlerArray)) - 1;
-GVAR(perFrameHandlerArray) = _newArray;
+params [["_handle", -1, [0]]];
+
+if (_handle < 0 || {_handle > GVAR(lastPFHid)}) exitWith {};
+
+[{
+    params ["_handle"];
+
+    GVAR(perFrameHandlerArray) deleteAt (GVAR(PFHhandles) select _handle);
+    GVAR(PFHhandles) set [_handle, nil];
+
+    {
+        _x params ["", "", "", "", "", "_handle"];
+        GVAR(PFHhandles) set [_handle, _forEachIndex];
+    } forEach GVAR(perFrameHandlerArray);
```

Comment:
Derp. The problem is that the __public__ handle is pushed to `GVAR(PFHhandles)` (which is for the internal handles, those who can change).
Thats why removing the PFH twice "works". The loop auto corrects the internal handle...